src/solve.jl

```diff
@@ -31,8 +33,9 @@ end
 function DiffEqBase.__solve(prob::BVProblem, alg::AbstractMIRK; dt = 0.0, abstol = 1e-3,
     adaptive::Bool = true, kwargs...)
     dt ≤ 0 && throw(ArgumentError("dt must be positive"))
-    T = eltype(prob.u0)
-    n = Int(cld((prob.tspan[2] - prob.tspan[1]), dt))
+    T = isa(prob.u0[1], Vector) ? eltype(prob.u0[1]) : eltype(prob.u0)
+    n = isa(prob.u0[1], Vector) ? (length(prob.u0) - 1) :
```

Comment:
Judging if the provided ```u0``` is an initial guess or the initial condition in this way is pretty awkward, do we have any other solutions to check the type of ```u0```?

Reply:
I don't think we have a better way.